### ISSUE TITLE
Add support for blocks that can return values

### ISSUE THREAD

**Issue by Sword-Smith** [TritonVM/tasm-lang#42]
We want to be able to write 

```rust
let a: u32 = {
    let c: u32 = 100;
    c
}
```

Supporting blocks that can return values would be a good way to prevent memory spilling, as the stack is cleared after the block. And we also need those expressions for a proper if/else expression implementation.

**Comment by Sword-Smith** [comment]
That means, we probably want to change the `BlockStmt` to 

```rust
BlockExpr {
    pub stmts: Vec<Stmt<T>>,
    pub return_expr: Option<<Expr<T>>
}
```

and then use this for while loops, blocks, if/else expressions, if/else statements etc.

**Comment by Sword-Smith** [comment]
Closed by 4877f00f6a8020af63fdde17387dc1a853138140
